server/image_list.go

```diff
@@ -70,3 +79,62 @@ func ConvertImage(from *storage.ImageResult) *types.Image {
 
 	return to
 }
+
+// FilterImageResults filters out the pinned images from the given list of image results
+func FilterPinnedImages(images []storage.ImageResult, pinnedImages []string) ([]storage.ImageResult, error) {
+	if len(pinnedImages) == 0 {
+		return images, nil
+	}
+	filteredImages := make([]storage.ImageResult, 0, len(images))
+	regexps, err := compileRegexpsForPinnedImages(pinnedImages)
```

Comment:
can these regexes be compiled on crio startup and stored so we don't need to recompile each time we list?

Reply:
they can be stored inside of the config similarly to how we do the seccomp profile or runtime handlers